Sources/SpeziHealthKit/Queries/HealthKitQuery.swift

```diff
@@ -0,0 +1,150 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import HealthKit
+
+
+extension HealthKit {
+    /// Run a one-off query.
+    ///
+    /// Use this function to perform a simple query of HealthKit samples.
+    ///
+    /// - parameter sampleType: The ``SampleType`` you want to fetch samples for
+    /// - parameter timeRange: The time range you want to fetch samples for.
+    /// - parameter limit: The number of objects that should be fetched. `nil` indicates that no limit should be applied.
+    /// - parameter filterPredicate: Optional refining predicate that allows you to filter which samples should be fetched.
+    public func query<Sample>(
+        _ sampleType: SampleType<Sample>,
+        timeRange: HealthKitQueryTimeRange,
+        limit: Int? = nil,
+        predicate filterPredicate: NSPredicate? = nil
+    ) async throws -> [Sample] {
+        let predicate = sampleType._makeSamplePredicate(
+            filter: NSCompoundPredicate(andPredicateWithSubpredicates: [timeRange.predicate, filterPredicate].compactMap(\.self))
+        )
+        let queryDescriptor = HKSampleQueryDescriptor<Sample>(
+            predicates: [predicate],
+            sortDescriptors: [SortDescriptor<Sample>(\.startDate, order: .forward)],
+            limit: limit
+        )
+        return try await queryDescriptor.result(for: healthStore)
+    }
+    
+    
+    /// Run an anchor query.
+    ///
+    /// Use this function to perform a one-off anchor query.
+    ///
+    /// You can use the `limit` parameter to implement a batched query:
+    /// ```swift
+    /// // This example fetches all heart rate samples
+    /// // of the past half year, 5000 samples at a time
+    /// var anchor = QueryAnchor<HKQuantitySample>()
+    /// while true {
```

Comment:
Not sure if a while `true` loop example is a great demonstration of the functionality, maybe rather add an example that stores the anchor somewhere in a type and offers a functionality to query the next samples or something similar?

Reply:
i thought about that as well, but the reason i went with this example specifically was that:
1. we don't have a dedicated batch query function, and this demonstrates that it's easy use the existing functions to get batching behaviour
2. it demonstrates that the anchors are lightweight objects which can just be created and used on-demand for something like this. IMP that also implies that you can store them and re-use them later on.

tbh, i do agree that we should have an example with that as well, maybe in the documentation of the `QueryAnchor` type itself?

Reply:
Sounds good; we can keep the example as it is 👍

Might be good to add a nice AsyncSequence for this use case in the future.